src/core/server/http/logging/get_payload_size.ts

```diff
@@ -25,9 +26,23 @@ const isFsReadStream = (src: unknown, res: Response): src is ReadStream => {
     res.source instanceof ReadStream
   );
 };
+const isZlibStream = (src: unknown, res: Response): src is Zlib => {
+  return (
+    !isBoom(res) &&
+    res.variety === 'stream' &&
+    res.source === src &&
+    typeof src === 'object' &&
+    src !== null &&
+    'bytesWritten' in src
+  );
+};
 const isString = (src: unknown, res: Response): src is string =>
   !isBoom(res) && res.variety === 'plain' && typeof src === 'string';
 
+const isObject = (src: unknown, res: Response) => {
```

Comment:
let's pass `res.variety` instead of the whole object to make sure nothing else is accessed